consul.go

```diff
@@ -104,24 +90,16 @@ func NewGroupReader(config GroupConfig) (Reader, error) {
 	return consulReader, nil
 }
 
-func (reader *consulReader) acquire(partitions int) (int, error) {
+func (reader *consulReader) acquire(ctx context.Context, partitions int) (int, error) {
 	// Cycle through each partition and try to acquire a lock to own it.
 	// Consul doesn't have support for sequential keys.
 	for partition := 0; partition < partitions; partition++ {
-		pair := &api.KVPair{
-			Key:     baseKey + reader.name + "/" + strconv.Itoa(partition),
-			Value:   []byte{},
-			Session: reader.session,
-		}
-
-		res, _, err := reader.client.KV().Acquire(pair, &api.WriteOptions{})
-		if err != nil {
-			return 0, err
+		lock, _ := consul.Lock(ctx, baseKey+reader.name+"/"+strconv.Itoa(partition))
```

Comment:
how do we specify the client or does it automatically grab the session's client from the context? @achille-roussel